[IMP] web: check permission for onchange call

kmagusiak opened the pull request:
Check that the user has access to the model when running `onchange` so that users that do not have access cannot use that feature. An exception is made for res.users where some fields are editable by the users themselves which are defined in SELF_WRITEABLE_FIELDS. This is probably something that we should change in the future.

odoo/enterprise#76342

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/192112.png)](https://mergebot.odoo.com/odoo/odoo/pull/192112)

kmagusiak commented:
@odoo/rd-security Could you review this PR?

I have added a check on the `onchange` function to allow it to work only on accessible records/models. Except for res.users where SELF_WRITEABLE_FIELDS overwrites the way it works.
Also there is a record rule that is overwritten which I removed since it does not show up anymore.

kmagusiak commented:
@mart-e I removed the commit you are fixing.
@xmo-odoo Is this ok as-is?

ryv-odoo reviewed (approved):
@robodoo r+

robodoo commented:
@kmagusiak @ryv-odoo linked pull request(s) odoo/enterprise#76342 not ready. Linked PRs are not staged until all of them are ready.
